screens/ActionsType.tsx

```diff
@@ -0,0 +1,219 @@
+
+
+  const getActionsByType = async (actionType: string) => {
+    if (!actionType) return;
+    const resp = await server.get(`${API.actions}/${actionType}`, {dispatch});
+    setResults(resp.data)
+    console.info('resp', JSON.stringify(resp.data));
+  }
+
+  const handleValueChange = (value) => {
+    if (selectedValues.some((selectedValue) => selectedValue.actionId === value.actionId)) {
+      setSelectedValues(selectedValues.filter((selectedValue) => selectedValue.actionId !== value.actionId));
+    } else {
+      setSelectedValues([...selectedValues, value]);
+    }
+  };
+  
+  const postAllActions = async () => {
+    let totalPoints = 0;
+    let totalUserPoints
+  
+    if (selectedValues) {
+      selectedValues.map( async (value, index) => {
+        await postAction(value.actionId);
+        totalPoints += value.actionPoints;
+        totalUserPoints = await addPoints(value.actionPoints);
+        console.log("points", totalUserPoints)
+
+        if (selectedValues.length === index + 1){
+          navigation.navigate("ActionsCongrats", {totalPoints, totalUserPoints})
+        }
+
+      });
+    }
+  };
+
+
+  const postAction = async (actionId: string) => {
+    const resp = await server.post(`${API.actions}/newaction`, {
+      actionId: actionId,
+
+    }, {dispatch})
+
+  }
+
+  const addPoints = async (actionPoints : number) => {
+    const resp = await server.post(`${API.user}/points`, {
+      points: actionPoints,
+      origin: "actions"
+    }, {dispatch})
+    const userPoints = resp.data.totalPoints
+    console.log(userPoints)
+    return userPoints
+  }
+
+  const goTo = (route) => {
+    navigation.navigate(route)
+  }
+
+  useEffect(() => {
+    getActionsByType(actionType)
+  }, [actionType])
+
+
+  const renderImage = () => {
+    switch (actionType) {
+      case 'transport':
+        return <Image 
+                    source={require('../assets/images/test.png')} 
+                    style={{justifyContent:'center'}}
+                    resizeMode='contain'
+                    alt="Transport Image"
+                />;
+      case 'food':
+        return <Image 
+                    source={require('../assets/images/food.png')} 
+                    style={{justifyContent:'center'}}
+                    resizeMode='contain'
+                    alt="Food Image"
+                />;
+      case 'energies':
+        return <Image 
+                    source={require('../assets/images/energies.png')} 
+                    style={{justifyContent:'center'}}
+                    resizeMode='contain'
+                    alt="Energies Image"
+               />;
+      case 'products':
+        return <Image 
+                    source={require('../assets/images/products.png')} 
+                    style={{justifyContent:'center'}}
+                    resizeMode='contain'
+                    alt="Products Image"
+                />;
+      default:
+        return null;
+    }
```

Comment:
This is completely fine. But, you might want to consider not repeating as much as possible, here is a suggestion:
```suggestion
  const renderImage = () => {
    const images = {
       transport: {
         name: "test",
         alt: "Transport Image",
       },
       food: {
         name: "food",
         alt: "Food Image",
       },
       energies: {
         name: "energies",
         alt: "Energies Image",
       },
       products: {
         name: "products",
         alt: "Products Image",
       },
    };
    const image = images[actionType];
    if (!image) return null;
    return <Image 
          source={require(`../assets/images/${image.name}.png`)} 
          style={{justifyContent:'center'}}
          resizeMode='contain'
          alt={image.alt}
      />;
    }
```

Reply:
I tried to implement this one but I got an error.
iOS Bundling failed 84ms
error: screens/ActionsType.tsx: screens/ActionsType.tsx:Invalid call at line 108: require("../assets/images/" + image.name + ".png")

I checked the documentation and I found this:

<img width="838" alt="Screen Shot 2023-03-16 at 4 45 54 PM" src="https://user-images.githubusercontent.com/99352268/225776647-c7346d4d-4265-4e73-a1aa-2ac010488908.png">

Can we leave it like this for now, and for the final I'll find as way to make it less repetitve.